pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -535,3 +537,19 @@ def teardown(self) -> None:
             self.lightning_module.cpu()
             # clean up memory
             torch.cuda.empty_cache()
+
+    def load_model_state_dict(self, checkpoint: Mapping[str, Any]) -> None:
+        if "state_dict" not in checkpoint and self._self_deleted_checkpoint_state_dict:
```

Comment:
Why do you need `self._self_deleted_checkpoint_state_dict` ? note: The variable name isn't great :)

Reply:
original PR [comment thread](https://github.com/PyTorchLightning/pytorch-lightning/pull/9585#discussion_r711380142):  I think in case `load_model_state_dict` gets called after `load_checkpoint`, @ananthsub was this the intention?

Reply:
@tchaton it's to distinguish if the 

@jjenniferdai - we should also ensure we reset the property in the `teardown` function

Reply:
"but later, the checkpoint connector will call `load_model_state_dict`" - this doesn't happen anymore after these checkpoint connector changes right?

Reply:
Sounds good to me, but let's find a better variable name. Something like `self._has_loaded_state_dict`